crypto/provider_core.c

```diff
@@ -816,6 +816,23 @@ int OSSL_PROVIDER_set_default_search_path(OSSL_LIB_CTX *libctx,
     return 0;
 }
 
+int OSSL_PROVIDER_get1_default_search_path(OSSL_LIB_CTX *libctx,
```

Comment:
Just a random thought but would `OSSL_PROVIDER_get0_default_search_path` be more sensible?  The caller gets to decide to strdup or not -- the path exists for the life of the libctx, so not duping it is reasonable.
```c
const char *OSSL_PROVIDER_get0_default_search_path(OSSL_LIB_CTX *libctx)
{
    char *path = NULL;

    if ((store = get_provider_store(libctx)) != NULL
            && CRYPTO_THREAD_read_lock(store->default_path_lock)) {
        path = OPENSSL_strdup(store->default_path ? store->default_path : "");
        CRYPTO_THREAD_unlock(store->default_path_lock);
    }
    return path;
}
```

Reply:
That is always an option; in that case, I would just do:
```
path = store->default_path;
```
You do get less error info that way; if the lock failed for some reason or the library context was invalid which is why I went the other route.

Reply:
Sorry, didn't look at the code properly :(

An invalid library context is likely to crash.  The lock failing is a legitimate concern I guess.

Reply:
I think an invalid library context should return NULL on `get_provider_store` (it calls `ossl_lib_ctx_get_data` which will also fail gracefully).

Reply:
or it will crash or return rubbish.  C has no way to distinguish a _good_ pointer from a _bad_ one.

Reply:
ok, switched to `OSSL_PROVIDER_get0_default_search_path`